### DISCUSSION TITLE
[ToolbarButton] Does v2 ToolbarButton support other events like onMouseEnter?

### DISCUSSION THREAD

**Discussion by Xidai** [UI5/webcomponents-react#6620] [Q&A] [answered]
We need to add onMouseEnter event to ToolbarButton, but it doesn't work. Is there any way to support it?

**Comment by Lukas742** [comment, marked as answer]
Hi @Xidai 

sine the `ToolbarButton` is an [abstract web component](https://sap.github.io/ui5-webcomponents-react/v2/?path=/docs/knowledge-base-faq--docs#what-are-abstract-ui5-web-components) native DOM events are not registered on the correct element in the shadow root. 
It is not recommended customizing the shadow-root of a component, because internal changes are not subject to semantic versioning, so I'd suggest opening a feature request in the [ui5-webcomponents repo](https://github.com/SAP/ui5-webcomponents/issues/new/choose) explaining your use-case.

If you need to enable this urgently, you could try adding a listener on the button inside the shadow-root, you can get the reference by calling `getDomRef()` on the `ui5-toolbar-button` element.

**Reply by Xidai** [reply]
I'll try that, thank you!
